### ISSUE TITLE
TripletNetwork lacks dataset shuffling causing overfitting and poor validation performance

### ISSUE THREAD

**Issue by VecherVhatuX** [VecherVhatuX/learning_data_science_in_prison#1348]
The current implementation of the TripletNetwork does not shuffle the dataset at the start of each epoch, leading to overfitting and poor performance on the validation set. The TripletDataset class does not provide a mechanism to shuffle the dataset at the start of each epoch.
